ansible/roles_ocp_workloads/ocp4_workload_dm7_foundations_elt/tasks/workload_dm.yml

```diff
@@ -0,0 +1,156 @@
+---
+
+- name: create image pull secret for registry.redhat.io
+  k8s:
+    state: present
+    namespace: "{{ ocp4_workload_dm7_foundations_elt_project_dm }}"
+    definition: "{{ lookup('file', role_path ~ '/files/image-pull-secret.yml' ) | from_yaml }}"
+
+- name: link secret to default sa
+  shell: oc secrets link default registry-credentials --for=pull -n {{ ocp4_workload_dm7_foundations_elt_project_dm }}
```

Comment:
Unless you need shell functions or variables, `command` module is typically preferred.